### ISSUE TITLE
Controlling redefinition behavior

### ISSUE THREAD

**Issue by dopplershift** [hgrecco/pint#1700]
Over in MetPy, in our past release we moved to use Pint's `get_application_registry()`. This had the [consequence](https://github.com/Unidata/MetPy/issues/2884) of changing the behavior of redefintion from `'warn'` to `'raise'`. I see that this behavior was intentionally added in 17fc8731abe2d01c8b5cb03cd1b83c12b6bd029e. Was this to make sure things didn't stomp on each other unintentionally? I think this makes sense, but just to ask: is this still the most sensible default behavior in a world with `get_application_registry()`?

Assuming this difference in defaults is the right answer, could we promote the `_on_redefinition` attribute to a public one so that there's a way within the supported API to control this behavior? The value of this attribute is only used at a few places when redefinitions occur, so it shouldn't be problematic to have people swapping it back and forth.

Right now I don't have any good work-arounds for our users.

**Comment by hgrecco** [comment]
> Was this to make sure things didn't stomp on each other unintentionally? 
Exactly

I agree with your suggestion of proming `_on_redefinition`  to a public one. But before doing that, is there any other attribute that has the same issue?

**Comment by dopplershift** [comment]
I don't believe so. `LazyRegistry` is only doing special handling for `_on_redefinition`, nothing else.
